src/net_processing.cpp

```diff
@@ -1866,7 +1866,9 @@ bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex)
 {
     AssertLockHeld(cs_main);
     if (m_chainman.ActiveChain().Contains(pindex)) return true;
-    return pindex->IsValid(BLOCK_VALID_SCRIPTS) && (m_chainman.m_best_header != nullptr) &&
+    // HaveNumChainTxs implies we at some point reached VALID_TRANSACTIONS, meaning
+    // we may have advertised it to peers via compact blocks
+    return pindex->HaveNumChainTxs() && (m_chainman.m_best_header != nullptr) &&
```

Comment:
Should adjust doc of `BlockRequestAllowed` above  ("we fully-validated them at some point")